### ISSUE TITLE
two differences between the implementation and the paper

### ISSUE THREAD

**Issue by xianghang** [BestActionNow/CVAR#4]
https://github.com/BestActionNow/CVAR/blob/e882d50417677ead608cc62276903a147b0b8f28/model/warm.py#L341-L360

AFAIU, this should correspond to equation (7) - (10) in the paper. However, there seem to be two discrepancies between the implementation and the paper
1. in eq (7) the prior encoder takes only item side information as input, but in the above implementation the variable `sideinfo_emb` includes _both_ side information embeddings and ID embeddings;
2. from the notations in the paper, the side information embedding $v_I$ in eq (7) seems to be part of $v_X$ in eq(1), which are the embeddings in the base model. However, in the above implementation, $v_I$ is from a separate embedding layer built only for the warm-up model, as in

https://github.com/BestActionNow/CVAR/blob/e882d50417677ead608cc62276903a147b0b8f28/model/warm.py#L289-L300

Is the above understanding correct? If so, what are the motivations for these two discrepancies?

**Comment by BestActionNow** [comment]
Yes, your comments are correct and these two discrepancies indeed exist. I'm sorry about that I haven't give clear explianation on

**Comment by BestActionNow** [comment]
Yes, your comments are correct and these two discrepancies indeed exist.

For the second question about different side-info embedding tables between the backbone and warm-up part, it dues to the deficient generalized ability of pretrained side-info embeddings in backbone model. We believe that there are two reasons for this phenomenon: 1. The numbers and categories of Side-info features are very limited in these two datasets. 2. With a single CTR training target, the pre-trained embeddings of side-info always have an overlarge influence on predicting the click label(such as the feature: year in movielens1M dataset) which makes it not suitable for predicting the latent ID representation. So we re-initialize a new emb table in the warm-up part.

For the first discrepancy you mentioned, firstly it also dues to lack of item features in datasets. What's more, in our industrial online training and predicting scene we found adding new initialized ID embedding will efficiently calibrate the learning process and accelerate the convergence, so we retained this trick in our implementation.

Thank you very much for your careful review and valuable comments on this work and I'm sorry for our oversight of these discrepancies.
